std/algorithm/sorting.d

```diff
@@ -2474,10 +2474,10 @@ schwartzSort(alias transform, alias less = "a < b",
 
     static double entropy(double[] probs) {
         double result = 0;
-        foreach (p; probs)
+        // ref works around https://issues.dlang.org/show_bug.cgi?id=16530
```

Comment:
A while ago I added a similar hack to std.entropy:

https://github.com/dlang/phobos/pull/4587/commits/11b4c1fb1eb4e1dd1ac32f43cc51a0453c4fae64